### ISSUE TITLE
Drop unnecessary flags

### ISSUE THREAD

**Issue by Massimo-B** [hartwork/resolve-march-native#44]
Is there any way to drop unnecessary flags that are implicitly set by -march anyway?
```
resolve-march-native --keep-identical-mtune --keep-default-params -a
-march=tigerlake -mabm -madx -maes -mavx -mavx2 -mavx512bitalg -mavx512bw -mavx512cd -mavx512dq -mavx512f -mavx512ifma -mavx512vbmi -mavx512vbmi2 -mavx512vl -mavx512vnni -mavx512vp2intersect -mavx512vpopcntdq -mbmi -mbmi2 -mclflushopt -mclwb -mcx16 -mf16c -mfma -mfsgsbase -mfxsr -mgfni -mlzcnt -mmmx -mmovbe -mmovdir64b -mmovdiri -mpclmul -mpku -mpopcnt -mprfchw -mrdpid -mrdrnd -mrdseed -msahf -msha -mshstk -msse -msse2 -msse3 -msse4.1 -msse4.2 -mssse3 -mtune=tigerlake -mvaes -mvpclmulqdq -mxsave -mxsavec -mxsaveopt -mxsaves --param=l1-cache-line-size=64 --param=l1-cache-size=48 --param=l2-cache-size=24576 -O2 -pipe
```
Are some of those options not already set implicity?

### Background
I need this tool to determine the explicit flags behind -march=native in order to pass them via distcc to remote servers that can't detect them.
Before I used different approaches like
```
# gcc -v -E -x c -march=native -mtune=native - < /dev/null 2>&1 | grep cc1 | perl -pe 's/^.* - //g;'
# gcc -### -march=native -x c -
```

**Comment by hartwork** [comment]
Hi @Massimo-B there is no way unless we teach that to the tool in a way that allows to encode that knowledge for all arches of interest. I'm not sure that work is worth anyone's time though. For your goal, does saving a few characters even matter enough? Are you interested in providing a clean and extensible implementation as a pull request?

**Comment by negril** [comment]
You can compare the output of `gcc -c -Q -march=tigerlake --help=target` and `gcc -c -Q -march=native --help=target` to get a list of additional flags set that are not in march.

**Comment by hartwork** [comment]
@negril that's an interesting idea.

**Comment by Massimo-B** [comment]
> You can compare the output of `gcc -c -Q -march=tigerlake --help=target` and `gcc -c -Q -march=native --help=target` to get a list of additional flags set that are not in march.

So this only works when running on the native platform to evaluate the `march=native`? Currently being on haswell I compared:
```
diff <(gcc -c -Q -march=haswell --help=target) <(gcc -c -Q -march=native --help=target)
12c12
<   -mabm                       		[disabled]
---
>   -mabm                       		[enabled]
16c16
<   -maes                       		[disabled]
---
>   -maes                       		[enabled]
```

**Comment by hartwork** [comment]
@Massimo-B absolutely, yes!

**Comment by Massimo-B** [comment]
Does that mean on this haswell platform with
```
# resolve-march-native --keep-identical-mtune
-march=haswell -mabm -maes -mavx -mavx2 -mbmi -mbmi2 -mcx16 -mf16c -mfma -mfsgsbase -mfxsr -mlzcnt -mmmx -mmovbe -mpclmul -mpopcnt -mrdrnd -msahf -msse -msse2 -msse3 -msse4.1 -msse4.2 -mssse3 -mtune=haswell -mxsave -mxsaveopt --param=l1-cache-line-size=64 --param=l1-cache-size=32 --param=l2-cache-size=8192
```
I could actually drop all but -march -mtune -mabm and -maes and have the same result?
This was for using distcc where I need explicit flags.

But it would also mean, that when not using distcc and I usually only have
`CFLAGS="-march=native -mtune=native"`
that I would not get -mabm -maes if not adding it manually?

**Comment by Massimo-B** [comment]
Does it make sense to use that information to offer a --minimal option for getting the minimal CFLAG list?
I know this is only cosmetical.

**Comment by hartwork** [comment]
@negril @Massimo-B there is pull request #48 now — could you give it a try?

**Comment by negril** [comment]
You need to fix LANG to C for this to work. Sorry I forgot to point that out earlier.
```
$ resolve-march-native --version
resolve-march-native 2.0.0
$ resolve-march-native
ERROR: Line '  -m128bit-long-double        \t\t[eingeschaltet]' not understood
$ LANG=C resolve-march-native
-march=znver3 -mno-mwaitx -mno-pku -mno-wbnoinvd -mshstk --param=l1-cache-line-size=64 --param=l1-cache-size=32 --param=l2-cache-size=512
```

**Comment by hartwork** [comment]
@negril good point, thanks for bringing it up! :+1:

**Comment by hartwork** [comment]
@negril fixed by PR #50 now, merged, 2.1.0 released to Git and PyPI, and bumped in Gentoo.
